python/pyspark/sql/pandas/serializers.py

```diff
@@ -201,9 +201,26 @@ class ArrowStreamArrowUDTFSerializer(ArrowStreamUDTFSerializer):
     Serializer for PyArrow-native UDTFs that work directly with PyArrow RecordBatches and Arrays.
     """
 
-    def __init__(self, table_arg_offsets=None):
+    def __init__(self, table_arg_offsets=None, arrow_cast=True):
```

Comment:
Let's enable arrow_cast by default for ArrowUDTFs (it's a new feature) so we don't need a flag here.